Pcap++/src/PcapFileDevice.cpp

```diff
@@ -114,6 +118,93 @@ int IFileReaderDevice::getNextPackets(RawPacketVector& packetVec, int numOfPacke
 	return numOfPacketsRead;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// SnoopFileReaderDevice members
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+SnoopFileReaderDevice::~SnoopFileReaderDevice()
+{
+	m_snoopFile.close();
+}
+
+bool SnoopFileReaderDevice::open()
+{
+	m_NumOfPacketsRead = 0;
+	m_NumOfPacketsNotParsed = 0;
+
+	m_snoopFile.open(m_FileName.c_str(), std::ifstream::binary);
+	snoop_file_header_t snoop_file_header;
+	m_snoopFile.read((char*)&snoop_file_header, sizeof(snoop_file_header_t));

```

Comment:
Here we should probably check if enough bytes were read:
```cpp
m_snoopFile.read((char*)&snoop_file_header, sizeof(snoop_file_header_t));
if (!m_snoopFile)
{
	PCPP_LOG_ERROR("Cannot read snoop file header for '" << m_FileName << "');
	m_snoopFile.close();
	return false;
}
```

Reply:
Resolved in 85cf49c